### PR TITLE
fix: add active category class in a sibiling $state. 

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#314

screen shot 2015-01-27 at 5 40 29 pm

### CODE REVIEW

**Comment by ceoaliongroo** [comment]
The test works locally, marked @wip because travis.

File: behat/features/meter.feature [hunk 1]
```diff
@@ -6,5 +6,11 @@ Feature: Meter
   @javascript @wip
   Scenario: Show all meters of the account
     Given I login with user "carlos"
-    When I visit "#/dashboard"
+    When I visit "/#/dashboard/1"
     Then I should see all the markers
+
+  @javascript @wip
```
Comment (amitaibu) [inline behat/features/meter.feature]: travis doesn't execute tests with ``@wip``

Comment (amitaibu) [inline behat/features/meter.feature]: remove the ``@wip``

File: client/app/scripts/controllers/menu.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ angular.module('negawattClientApp')
     $scope.user = profile.user;
     $scope.timedate = Timedate;
 
+    console.log($state);
```
Comment (amitaibu) [inline client/app/scripts/controllers/menu.js]: remove

File: client/app/views/dashboard/main.categories.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
       <ol ui-tree-nodes="" ng-model="item.children" >
         <li ng-repeat="subItem in item.children" ui-tree-node ng-show="hasMeters(subItem)">
           <div ui-tree-handle class="tree-level-middle">
-            <a ui-sref="dashboard.withAccount.preload.categories({categoryId: subItem.id})" ui-sref-active="active-category">
+            <a ui-sref="dashboard.withAccount.preload.categories({categoryId: subItem.id})" ui-sref-active="active-category" ng-class="{'active-category':  ('categoryId' | includedByParams:subItem.id) }">
```
Comment (amitaibu) [inline client/app/views/dashboard/main.categories.html]: ng-class="{ active: $state.includes('dashboard.withAccount.preload.categores') || $state.includes('dashboard.withAccount.preload.markers')  }">

**Comment by ceoaliongroo** [comment]
screen shot 2015-01-28 at 12 33 12 pm

File: client/app/scripts/filters/uiRouterFilters.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * @name ui.router.state.filter:includedByParams
+ *
+ * @requires ui.router.state.$stateParams
+ *
+ * @description
+ * Translates to {@link ui.router.state.$state#methods_includes $state.includes('fullOrPartialStateName')}.
+ */
+
+angular.module('negawattClientApp')
+  .filter('includedByParams', function ($state, $stateParams) {
```
Comment (amitaibu) [inline client/app/scripts/filters/uiRouterFilters.js]: remove

File: client/app/scripts/filters/uiRouterFilters.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    var includesFilter = function (param, expression) {
+      return (angular.isDefined($stateParams[param]) && $stateParams[param] && +$stateParams[param] === expression);
+    };
+    includesFilter.$stateful = true;
```
Comment (amitaibu) [inline client/app/scripts/filters/uiRouterFilters.js]: change

**Comment by amitaibu** [comment]
Needs re-roll

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**